angular2-twitter/public/src/app/data-creator.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class DataCreatorService {
+
+  constructor() { }
+
+  createData(sortedData: Object, colors: string[]) {
+    let i = 0;
+    let data = {
+      bar: [],
+      map: [],
+      linear: []
+    }
+    Object.entries(sortedData).forEach(([key, value]) => {
+      data.bar = this.createBarData(data.bar, key, value, colors[i]);
+      data.map = this.createMapData(data.map, key, value, colors[i]);
+      data.linear = this.createLinearData(data.linear, key, value, colors[i]);
+      i++;
+    });
+    return data;
+  }
+
+  createBarData(bar: Object[], key: string, value: any[], color: string): Object[] {
+    //create JSON for Bar Graph {'word': 'text', 'freq': stream.length, 'color': 'color'}
+    bar.push({word: key, freq: value.length, color: color});
```

Comment:
złą praktyką jest modyfikowanie obiektu (również arrayki) przekazanego w argumencie funkcji. ze względu na to, że de facto nie musisz wtedy go zwracać, już został zmodyfikowany obiekt przekazany przez referencję